### ISSUE TITLE
Infinite recursion with GuildChannel#clone and/or Util.mergeDefault

### ISSUE THREAD

**Issue by rei2hu** [discordjs/discord.js#4602]
**Please describe the problem you are having in as much detail as possible:**

Util.mergeDefault with a circular object for both default and given creates infinite recursion. GuildChannel#clone accepts a ChannelResolvable for options.parent and the default can also be a channel object which can cause this problem.

Workaround is just to pass the channel's id instead.

https://github.com/discordjs/discord.js/blob/master/src/structures/GuildChannel.js#L508-L522
https://github.com/discordjs/discord.js/blob/master/src/util/Util.js#L292-L293

**Include a reproducible code sample here, if possible:**

```js
// assume c1 has a parent
const c1 = client.channels.cache.find(ch => ch.name === '1hu');
const c2 = await c1.clone({
  name: '2hu',
  parent: c1.parent,
});
```

**Further details:**

- discord.js version: master
- Node.js version: dunno
- Operating system: dunno
- Priority this issue should have – please be realistic and elaborate if possible: low-med

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by NotSugden** [comment]
> discord.js version: dunno
> Node.js version: dunno
> Operating system: dunno
you linked discord.js version 12.2.0 above
you can check your node version by using the `node -v` command
how do you not know your operating system?

**Comment by rei2hu** [comment]
My bad, I should've elaborated on that - this was a problem someone else encountered whose cause was found after looking through the library's source. I didn't take any information about the user's setup and didn't manually test it myself since I found the problem in the code.

Knowing the problem, I can say these details aren't very important to the issue except for maybe the library version which might as well be master but could also be several older ones depending on how far back this was implemented.
